From: Simon McVittie <smcv@debian.org>
Date: Fri, 22 Jun 2018 01:10:55 +0100
Subject: avahi: Fail immediately if we can't talk to D-Bus or Avahi

We special-case AVAHI_ERR_NO_DAEMON to not cause warnings, but if
we pass AVAHI_CLIENT_NO_FAIL to avahi_client_new, we never actually
see AVAHI_ERR_NO_DAEMON. Instead, we will get AVAHI_ERR_BAD_STATE
when we try to use the client.

Bug: https://github.com/ostreedev/ostree/issues/1618
Signed-off-by: Simon McVittie <smcv@debian.org>
Forwarded: https://github.com/ostreedev/ostree/pull/1639
Applied-upstream: 2018.7, commit:https://github.com/ostreedev/ostree/commit/e120a6b1198aaa785533c76316898f15a804dae1
---
 src/libostree/ostree-repo-finder-avahi.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/libostree/ostree-repo-finder-avahi.c b/src/libostree/ostree-repo-finder-avahi.c
index 223d8f0..bc38376 100644
--- a/src/libostree/ostree-repo-finder-avahi.c
+++ b/src/libostree/ostree-repo-finder-avahi.c
@@ -1432,8 +1432,7 @@ ostree_repo_finder_avahi_start (OstreeRepoFinderAvahi  *self,
 
   g_assert (self->client == NULL);
 
-  client = avahi_client_new (avahi_glib_poll_get (self->poll),
-                             AVAHI_CLIENT_NO_FAIL,
+  client = avahi_client_new (avahi_glib_poll_get (self->poll), 0,
                              client_cb, self, &failure);
 
   if (client == NULL)
